### DISCUSSION TITLE
Why does ImageMagick (liblqr) need OpenSSL?

### DISCUSSION THREAD

**Discussion by mcandre** [ImageMagick/ImageMagick#7021] [Development]
### ImageMagick version

7.1.1-26

### Operating system, version and so on

MacOS Sonoma

### Description

I was running `brew doctor` to look for oddities on my Mac, when I noticed that an older version of OpenSSL was installed.

I went to remove it, but Homebrew pointed out that another package was using it.

```console
$ brew remove openssl@1.1
Error: Refusing to uninstall /opt/homebrew/Cellar/openssl@1.1/1.1.1w
because it is required by liblqr, which is currently installed.
You can override this and force removal with:
  brew uninstall --ignore-dependencies openssl@1.1
```

So I went to remove that, but Homebrew pointed out that ImageMagick was using that.

```console
$ brew remove liblqr
Error: Refusing to uninstall /opt/homebrew/Cellar/liblqr/0.4.2_1
because it is required by imagemagick, which is currently installed.
You can override this and force removal with:
  brew uninstall --ignore-dependencies liblqr
```

I can understand why ImageMagick may need some kind of abstract scaling library dependency. But why does an image editor need an old version of OpenSSL? Why does it need OpenSSL at all?

How about updating the liblqr dependency, or modifying the Homebrew formula to disable OpenSSL for it?

Or perhaps releasing ImageMagick as a cask, so that it gets more of its dependencies baked in statically?

**Comment by urban-warrior** [comment]
ImageMagick does not necessitate OpenSSL. The issue might stem from one of the delegate library dependencies or could be related to HomeBrew.

**Comment by BetterB0y** [comment]
Here is solution to remove openssl@1.1 and still using ImageMagick
https://github.com/carlobaldassi/liblqr/issues/6#issuecomment-1938374197
